packages/components/src/tab-panel/index.tsx

```diff
@@ -0,0 +1,40 @@
+import styled, { StyledComponent } from '@emotion/styled';
+import { TabPanel as CoreTabPanel } from '@wordpress/components';
+
+export interface Tab extends Omit< CoreTabPanel.Tab, 'title' > {
+	title: React.ReactChild;
+}
+
+export interface TabPanelProps extends Omit< CoreTabPanel.Props, 'children' | 'tabs' > {
+	tabs: readonly Tab[];
+	children: ( tab: Tab ) => JSX.Element;
+}
+
+/**
+ * A wrapper around WordPress's own <TabPanel> component.
+ *
+ * - Adjust style so it aligns with what we want at Automattic
+ * - Allows arbitrary elements in the tab titles (this is already possible with the
+ *   WordPress component even though the documentation says only strings can be used).
+ */
+export const TabPanel: StyledComponent< TabPanelProps > = styled( CoreTabPanel )`
```

Comment:
Does it make sense to export styled versions of generic, reusable components? If someone on Jetpack Cloud wants to reuse the logic, they'd need to apply different styles anyway, so is it worth it to keep it here? I'd prefer to have it scoped where it's used in SMD.

Reply:
That's fair enough. I was trying to take some inspiration from @SaxonF who I think intended the tab control he mocked up to be universal. So Jetpack Cloud _would_ reuse it as is. That's why I tried to keep all the special positioning rules in the SMD.

Of course it's colour, internal padding, etc. are slightly different from the existing universal WordPress tab control 😄 so I thought of this as the "a8c style" one.

But I think you're right, I might be optimistic in thinking of this as the one true a8c style tab component. What if I moved it to `packages/components/sites-table/sites-table-tab-panel.tsx` and called it `SitesTableTabPanel`?

Reply:
Then it wouldn't be as generic as we thought 😅  Maybe we could export the style object and then the consumers might optionally use it? Or, on second thought, we could export in `packages` the un-styled version and in Calypso create a new version with the styles on top of it. JC and SMD are still in Calypso...

Reply:
Follow up https://github.com/Automattic/wp-calypso/pull/65876